### ISSUE TITLE
Errors not shown when using multiple validators with async validator

### ISSUE THREAD

**Issue by dcbacarro** [joanpablo/reactive_forms#249]
I have an input that accepts a mobile number and I have the following validators set:

```
'mobile_no': FormControl<String>(
      asyncValidatorsDebounceTime: 1000,
      validators: [
        Validators.required,
        Validators.pattern(MOBILE_NO_PATTERN),
      ],
      asyncValidators: [checkMobile],
    ),
```

The MOBILE_NO_PATTERN value is `r'^(09)\d{9}$'`
The checkMobile async validator sends a request to the server to check if the mobile number already exists in the database.

When I enter (by typing quite quickly) an invalid mobile number, the async validator is triggered (because at some point while typing, the mobile number matches the pattern). The async validator returns valid but the value is not validated again for the pattern.

I tried specifying a longer async debounce timeout but the unexpected behavior still occurs.

I am using reactive_forms ^10.6.0

**Comment by joanpablo** [comment]
Hi @dcbacarro,

Thanks for using **Reactive Forms** and for the issue. Could you show me the implementation of the **checkMobile** function?

**Comment by dcbacarro** [comment]
You're welcome @joanpablo. It really made it easier to create forms with this package in Flutter.

Here's the implementation of the **checkMobile** function based on the documentation:

```
Future<Map<String, dynamic>?> checkMobile(
    AbstractControl<dynamic> control) async {
  final error = {'unique': false};
  try {
    final res = await validateMobile(control.value);
    if (res!.data!['invalid'] == true) {
      control.markAsTouched();
      return error;
    }
  } catch (e, stack) {
    if (ENABLE_SENTRY) await Sentry.captureException(e, stackTrace: stack);
    control.markAsTouched();
    return error;
  }

  return null;
}
```

**Comment by joanpablo** [comment]
Have you tried the *complex* example that is in the **example** folder of the **Reactive Forms** project? In that example, we show a very similar use case, but with an *email*. Can you check this example and let us know if it has the desired behavior?

**Comment by dcbacarro** [comment]
> Have you tried the _complex_ example that is in the **example** folder of the **Reactive Forms** project? In that example, we show a very similar use case, but with an _email_. Can you check this example and let us know if it has the desired behavior?

Okay, I'll have a look at the complex example. Will get back to you later. Thanks @joanpablo.

**Comment by dcbacarro** [comment]
I think the complex example is the same with the example shown in the readme. Am I right? If so, as I mentioned in my comment [here](https://github.com/joanpablo/reactive_forms/issues/249#issuecomment-999511623), that is where I based my implementation :)

**Comment by joanpablo** [comment]
Perfect meantime let me recap, to see if I understand your issue:
1- Type correct phone number
2- Patter validation return OK
3- Async Validation Trigger the request (this operation take some time)
4- Type another character and Pattern Validation marks the control as **invalid**
5- The Async Validation responds and marks the control as Valid.
6- The result of the control is marked as valid even when the phone number is wrong.

Is that the use case?

**Comment by dcbacarro** [comment]
> Perfect meantime let me recap, to see if I understand your issue: 1- Type correct phone number 2- Patter validation returns OK 3- Async Validation Trigger the request (this operation take some time) 4- Type another character and Pattern Validation marks the control as **invalid** 5- The Async Validation responds and marks the control as Valid. 6- As a result, the control is marked as valid even when the phone number is wrong.
> 
> Is that the use case?

Yes, correct. #5 and #6 only occurs when the user types quite quickly. I'll have a screen recording of the behavior. Please wait for a moment. :)

**Comment by dcbacarro** [comment]
Hi @joanpablo, here's the recording. Please notice that when I type normally, the validations are working properly. Also when I type after the loading animation disappears the validation for the pattern does not appear but it appears on blur.

![ezgif-3-fc04ec9ee7](https://user-images.githubusercontent.com/49216769/147089312-8a731c7c-8ba2-4c3f-adca-406334aa1b42.gif)
.

**Comment by joanpablo** [comment]
Hi @dcbacarro,

Can you update to the last version of **Reactive Forms** and see if the issue persists?
The complex example in the **example** folder works perfectly as expected and is quite similar to your use case. I will try to reproduce your issue with your Pattern mobile number, and see what happend.  

But could you update the version?

**Comment by dcbacarro** [comment]
Sure, I'll try to update to the latest version. I'll let you know if the issue still persists after updating. Thanks @joanpablo.

**Comment by joanpablo** [comment]
Hi @dcbacarro,

I was able to reproduce the issue, and a solution is coming :).

I still need to create several tests before releasing the solution. I need to be sure nothing else is broken with the changes. So I hope that by the end of today I could release a new version with this issue solved.

Thanks for the issue.

**Comment by dcbacarro** [comment]
Hi @joanpablo, I have updated the version to **v10.6.7** and this issue still occurs. As a workaround for now, I call `updateValueAndValidity()` on the control upon blur.

**Comment by dcbacarro** [comment]
> Hi @dcbacarro,
> 
> I was able to reproduce the issue, and a solution is coming :).
> 
> I still need to create several tests before releasing the solution. I need to be sure nothing else is broken with the changes. So I hope that by the end of today I could release a new version with this issue solved.
> 
> Thanks for the issue.

Really looking forward to this. Thank you very much @joanpablo!

**Comment by joanpablo** [comment]
Hi again @dcbacarro,

I have just released the version v10.6.8 that solved the issue. Please check it, and let me know if everything is working correctly.

I will close the issue but if you have not solved the issue please re-open it, or create a new one.

Thanks again.

**Comment by dcbacarro** [comment]
Thanks @joanpablo. Confirming that this issue is fixed now. Cheers!
